terraform/kubelet-private-registry/main.tf

```diff
@@ -0,0 +1,20 @@
+terraform {
+  required_providers {
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "~> 2.30"
+    }
+  }
+}
+
+resource "kubernetes_manifest" "kubelet-private-registry-secret" {
+  computed_fields = ["stringData"]
+  manifest = yamldecode(templatefile("${path.module}/manifests/kubelet_private_registry-secret.yaml.tpl", {
+    "image_registry_url"  = var.image_registry_url
+    "image_registry_auth" = base64encode("${var.image_registry_username}:${base64decode(var.image_registry_password)}")
+  }))
+}
```

Comment:
Will this be printed in the terraform plan in clear text? If so, then it could be better not to use the generic manifest resource but the `core/kubernetes_secret` resource instead. We do this in other project already:

```
resource "kubernetes_secret" "docker-registry" {
  metadata {
    name = "my-pull-secret"
    namespace = var.server_namespace
  }

  data = {
    ".dockerconfigjson" = templatefile("${path.module}/pullSecret.json.tftpl", {
      server = var.my_pull_secret_server
      username = var.my_pull_secret_username
      password = var.my_pull_secret_password
      auth = base64encode("${var.my_pull_secret_username}:${var.my_pull_secret_password}")
    })
  }

  type = "kubernetes.io/dockerconfigjson"
}
```

and a template file like:
```
{"auths":{"${server}":{"username":"${username}","password":"${password}", "auth":"${auth}"}}}
```

Reply:
I changed it to use the "kubernetes_secret"-resource and also added the ability to configure multiple registries